stream.go

```diff
@@ -45,11 +48,36 @@ Loop:
 			break Loop
 		}
 
-		reader = bufio.NewReader(stream.getResponse().Body)
+		resp := stream.getResponse()
+		if resp.StatusCode != 200 {
+			gitter.log(fmt.Sprintf("Unexpected response code %v", resp.StatusCode))
+			continue
+		}
+
+		//"The JSON stream returns messages as JSON objects that are delimited by carriage return (\r)" <- Not true crap it's (\n) only
+		reader = bufio.NewReader(resp.Body)
 		line, err := reader.ReadBytes('\n')
-		if stream.isClosed() {
+
+		//Check if the line only consists of whitespace
+		onlyWhitespace := true
+		for _, b := range line {
+			if b != ' ' && b != '\t' && b != '\r' && b != '\n' {
+				onlyWhitespace = false
+			}
+		}
+
+		if onlyWhitespace {
+			//"Parsers must be tolerant of occasional extra newline characters placed between messages."
+			currentKeepalive := time.Now().Unix() //interesting behavior of 100+ keepalives per seconds was observed
+			if currentKeepalive-lastKeepalive > 10 {
+				gitter.log("Keepalive was received")
+			}
+			continue
```

Comment:
The idea of this label is to be able breaking the loop no matter how deep you are in the inner loops or switch-cases. Means, you can break the the main loop even if you are inside switch case. Check this sample https://golang.org/doc/effective_go.html#switch (scroll down a little :))